src/Renci.SshNet/Sftp/SftpFileStream.cs

```diff
@@ -838,29 +818,28 @@ public override long Seek(long offset, SeekOrigin origin)
                     // Abandon the read buffer.
                     _bufferPosition = 0;
                     _bufferLen = 0;
+                }
 
-                    // Seek to the new position.
-                    switch (origin)
-                    {
-                        case SeekOrigin.Begin:
-                            newPosn = offset;
-                            break;
-                        case SeekOrigin.Current:
-                            newPosn = _position + offset;
-                            break;
-                        case SeekOrigin.End:
-                            var attributes = _session.RequestFStat(_handle, false);
-                            newPosn = attributes.Size - offset;
-                            break;
-                    }
-
-                    if (newPosn < 0)
-                    {
-                        throw new EndOfStreamException();
-                    }
+                // Seek to the new position.
+                switch (origin)
+                {
+                    case SeekOrigin.Begin:
+                        newPosn = offset;
+                        break;
+                    case SeekOrigin.Current:
+                        newPosn = _position + offset;
+                        break;
+                    case SeekOrigin.End:
+                        var attributes = _session.RequestFStat(_handle, false);
+                        newPosn = attributes.Size + offset;
+                        break;
+                }
 
-                    _position = newPosn;
+                if (newPosn < 0)
+                {
+                    throw new EndOfStreamException("End of stream.");
```

Comment:
You are providing a non-default exception message, causing the test SeekShouldHaveThrownEndOfStreamException to fail. I beleive if you just use ```throw new EndOfStreamException();``` this will be fixed.

Reply:
Thanks for pointing that out. The exception message is already present in another part of the code, so I assumed I should leave it in there.

Reply:
@drieseng There have been suggestions in other issues that we should localize exception messages. While that is probably too much work at this time, I propose to use default exception messages where possible and to stop checking error messages in the tests and rely on only asserting exception type (and other structured info in the exception) in the tests. Do you agree?

Reply:
I have noticed that the exception message format is different across .NET versions. For example:
.NET Framework 3.5
```
ArgumentException error message.
Parameter name: ParamName
```
.NET Core 3.0
```
ArgumentException error message. (Parameter 'ParamName')
```
This causes some tests to pass on one framework and fail on another.